service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionService.java

```diff
@@ -238,6 +238,19 @@ public void addEmptyDirectionsForAssigneeNotInMap(Map<DirectionAssignee, List<El
             .collect(toList());
     }
 
+    /**
+     * Removes directions where directionNeeded is set to No. Does not remove custom directions.
+     *
+     * @param directions any list of directions.
+     * @return a list of directions that are marked as needed.
+     */
+    public List<Element<Direction>> removeUnnecessaryDirections(List<Element<Direction>> directions) {
+        return directions.stream()
+            .filter(element -> "Yes".equals(element.getValue().getDirectionNeeded()) || "Yes".equals(
```

Comment:
Maybe refactor out the condition into its own method so that it is a bit easier to read then the flow in this method could just be
```java
        return directions.stream()
            .filter(this::showDirection)
            .collect(toList());
```

Reply:
👍